tokio/src/io/driver/mod.rs

```diff
@@ -251,7 +230,16 @@ impl Park for Driver {
         Ok(())
     }
 
-    fn shutdown(&mut self) {}
+    fn shutdown(&mut self) {
+        if self.inner.shutdown() {
+            self.resources.for_each(|io| {
```

Comment:
If this statement is commented out, do tests fail? i.e. if there is a memory leak, is it caught my CI? If not, we should add one.

Reply:
Yes. 

First, we have to comment out [clear_wakers()](https://github.com/hidva/tokio/blob/43fce8970e483138ebdca57bd9af32e115d007bd/tokio/src/io/driver/registration.rs#L209).

Second, if we commented out `io.shutdown()` here, we'd get a memory leak, otherwise we wouldn't.

```
==756352== LEAK SUMMARY:
==756352==    definitely lost: 96 bytes in 1 blocks
==756352==    indirectly lost: 20,856 bytes in 35 blocks
==756352==      possibly lost: 0 bytes in 0 blocks
==756352==    still reachable: 9,952 bytes in 69 blocks
==756352==         suppressed: 0 bytes in 0 blocks
==756352==
==756352== For lists of detected and suppressed errors, rerun with: -s
==756352== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
```

(It looks like we don't need to call `clear_wakers()` in `Drop for Registration ` anymore.